### DISCUSSION TITLE
GitHub to connect vercel

### DISCUSSION THREAD

**Discussion by ankush9131** [vercel/next.js#47460] [Help]
### Summary

Dear Team

When we are deploying Vercel app project from github repository then we are facing some error which we have shown below

### Additional information

```js
The provided GitHub repository does not contain the requested branch or commit reference. Please ensure the repository is not empty.
```

**Comment by sefin23** [comment]
This error means Vercel can't find the branch you're trying to deploy. Try these fixes:

Make sure your code is actually pushed — Run git push origin main in your terminal first. An empty or unpushed repo causes this exact error.

Check the branch name — Vercel defaults to main. If your repo uses master, go to Vercel Dashboard → your project → Settings → Git → change the Production Branch name.

Reconnect the GitHub repo — Go to Vercel Dashboard → Settings → Git → disconnect and reconnect your repository.

Check Vercel's GitHub app permissions — Go to GitHub → Settings → Applications → Vercel → Repository Access, and make sure your repo is listed.
After any of these, trigger a fresh deployment from the Vercel dashboard. Hope this helps!

**Comment by choueibecheikh-png** [comment]
[CGI-Fr-2023.pdf](https://github.com/user-attachments/files/32421526/CGI-Fr-2023.pdf)

**Comment by choueibecheikh-png** [comment]
<!doctype html>
<html lang="ar" dir="rtl">
<head>
  <meta charset="UTF-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>Concours Mohammed</title>
</head>
<body>
  <h1>مرحبًا بك في موقع Concours Mohammed</h1>
  <p>الموقع قيد الإنشاء.</p>
</body>
</html>
